diff --git a/src/helpers/browser_info.ts b/src/helpers/browser_info.ts
index 8531f64f..560b6ddb 100644
--- a/src/helpers/browser_info.ts
+++ b/src/helpers/browser_info.ts
@@ -68,6 +68,11 @@ export class BrowserInfo {
             case threema.BrowserName.Safari:
                 description = 'Safari ' + this.version;
                 break;
+            case threema.BrowserName.win32Desktop:
+            case threema.BrowserName.linuxDesktop:
+            case threema.BrowserName.macosDesktop:
+                description = 'Threema for Desktop ' + this.version;
+                break;
         }
         if (this.mobile) {
             description += ' [Mobile]';
diff --git a/src/partials/welcome.ts b/src/partials/welcome.ts
index 03a58b31..72ee154a 100644
--- a/src/partials/welcome.ts
+++ b/src/partials/welcome.ts
@@ -174,6 +174,7 @@ class WelcomeController {
                 this.log.warn('Edge is too old (' + version + ' < ' + minVersions.EDGE + ')');
                 this.showBrowserWarning();
             }
+        } else if (this.browser.name === threema.BrowserName.win32Desktop || this.browser.name === threema.BrowserName.macosDesktop || this.browser.name === threema.BrowserName.linuxDesktop) {
         } else {
             this.log.warn('Non-supported browser, please use Chrome, Firefox, Edge or Opera');
             this.showBrowserWarning();
diff --git a/src/services/browser.ts b/src/services/browser.ts
index c65b1541..419ea3a6 100644
--- a/src/services/browser.ts
+++ b/src/services/browser.ts
@@ -45,11 +45,17 @@ export class BrowserService {
                 edge: false,
                 opera: false,
                 safari: false,
+                macosDesktop: false,
+                win32Desktop: false,
+                linuxDesktop: false,
             };
 
             const uagent = this.$window.navigator.userAgent.toLowerCase();
 
-            browser.chrome = /webkit/.test(uagent) && /chrome/.test(uagent) && !/edg/.test(uagent);
+            browser.macosDesktop = /darwinthreemadesktop/.test(uagent) 
+            browser.win32Desktop = /win32threemadesktop/.test(uagent) 
+            browser.linuxDesktop = /linuxthreemadesktop/.test(uagent) 
+            browser.chrome = /webkit/.test(uagent) && /chrome/.test(uagent) && !/edg/.test(uagent) && !browser.macosDesktop && !browser.win32Desktop && !browser.linuxDesktop;
             browser.chromeIos = /mozilla/.test(uagent) && /crios/.test(uagent);
             browser.firefox = /mozilla/.test(uagent) && /firefox/.test(uagent);
             browser.firefoxIos = /mozilla/.test(uagent) && /fxios/.test(uagent);
@@ -80,6 +86,12 @@ export class BrowserService {
                         b = 'crios';
                     } else if (x === 'safari') {
                         b = 'version';
+                    } else if (x === 'macosDesktop') {
+                      b = 'darwinthreemadesktop'
+                    } else if (x === 'win32Desktop') {
+                      b = 'win32threemadesktop'
+                    } else if (x === 'linuxDesktop') {
+                      b = 'linuxthreemadesktop'
                     } else {
                         b = x;
                     }
@@ -125,6 +137,16 @@ export class BrowserService {
                 this.browser = new BrowserInfo(uagent, BrowserName.Opera, version);
             }
 
+            if (browser.win32Desktop) {
+                this.browser = new BrowserInfo(uagent, BrowserName.win32Desktop, version);
+            }
+            if (browser.macosDesktop) {
+                this.browser = new BrowserInfo(uagent, BrowserName.macosDesktop, version);
+            }
+            if (browser.linuxDesktop) {
+                this.browser = new BrowserInfo(uagent, BrowserName.linuxDesktop, version);
+            }
+
             // Detection failed
             if (this.browser === undefined) {
                 this.log.warn(`Browser could not be recognized. User agent: ${uagent}`);
diff --git a/src/threema.d.ts b/src/threema.d.ts
index eff12916..2a62fbfa 100644
--- a/src/threema.d.ts
+++ b/src/threema.d.ts
@@ -539,6 +539,9 @@ declare namespace threema {
         Edge = 'edge',
         Opera = 'opera',
         Safari = 'safari',
+        macosDesktop = 'macosThreemaDesktop',
+        win32Desktop = 'win32ThreemaDesktop',
+        linuxDesktop = 'linuxThreemaDesktop',
     }
 
     interface PromiseRequestResult<T> {
